gensim/matutils.py

```diff
@@ -725,15 +725,15 @@ def unitvec(vec, norm='l2', return_norm=False):
                 return vec
         else:
             if return_norm:
-                return vec, 1.
+                return vec, 1.0
             else:
                 return vec
 
     if isinstance(vec, np.ndarray):
         if norm == 'l1':
             veclen = np.sum(np.abs(vec))
         if norm == 'l2':
-            veclen = blas_nrm2(vec)
+            veclen = blas_nrm2(vec) if len(vec) else 0.0
```

Comment:
`len` on numpy arrays is ambiguous… better use `shape`?

Reply:
I replaced `if len(vec)` with `if 0 in vec.shape`.